#  Hypothesis Testing – Introduction

To better understand key patterns and relationships in our **Hybrid Manufacturing System (HMS)** dataset, we perform **statistical hypothesis testing**. This helps validate whether observed differences or correlations in the data are statistically significant or just due to random variation.

The dataset includes information about machine operations, processing times, energy consumption, scheduling accuracy, and optimization strategies. By testing specific hypotheses, we aim to answer questions such as:

- Do different **optimization strategies** lead to significantly different energy consumption?
- Are **processing times** significantly different between operation types like Lathe and Grinding?
- Does the **machine used** affect the likelihood of **job delays**?
- Is there a **correlation** between energy consumption and processing time?

The results from these tests can support data-driven decisions for improving efficiency, reducing delays, and optimizing energy use in the production system.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
df = pd.read_csv("hybrid_system_cleaned.csv")
df.head()

Job_ID Machine_ID Operation_Type  Material_Used  Processing_Time  \
0   J001        M01       Grinding           3.17               76   
1   J002        M01       Grinding           3.35               79   
2   J003        M04       Additive           2.29               56   
3   J004        M04       Grinding           1.76              106   
4   J005        M01          Lathe           1.90               46   

   Energy_Consumption  Machine_Availability      Scheduled_Start  \
0               11.42                    96  2023-03-18 08:00:00   
1                6.61                    84  2023-03-18 08:10:00   
2               11.11                    92  2023-03-18 08:20:00   
3               12.50                    95  2023-03-18 08:30:00   
4                8.13                    88  2023-03-18 08:40:00   

         Scheduled_End         Actual_Start           Actual_End Job_Status  \
0  2023-03-18 09:16:00  2023-03-18 08:05:00  2023-03-18 09:21:00  Completed   
1  2023-03-18 09:29:00  2023-03-18 08:20:00  2023-03-18 09:39:00    Delayed   
2  2023-03-18 09:16:00                  NaN                  NaN     Failed   
3  2023-03-18 10:16:00  2023-03-18 08:35:00  2023-03-18 10:21:00  Completed   
4  2023-03-18 09:26:00  2023-03-18 08:42:00  2023-03-18 09:28:00  Completed   

  Optimization_Category  Delay_Minutes  Scheduled_Duration  Actual_Duration  \
0   Moderate Efficiency            5.0                76.0             76.0   
1        Low Efficiency           10.0                79.0             79.0   
2        Low Efficiency            NaN                56.0              NaN   
3   Moderate Efficiency            5.0               106.0            106.0   
4       High Efficiency            2.0                46.0             46.0   

  Scheduled_Date Actual_Date  
0     2023-03-18  2023-03-18  
1     2023-03-18  2023-03-18  
2     2023-03-18         NaN  
3     2023-03-18  2023-03-18  
4     2023-03-18  2023-03-18

In [6]:
df.head()

Job_ID Machine_ID Operation_Type  Material_Used  Processing_Time  \
0   J001        M01       Grinding           3.17               76   
1   J002        M01       Grinding           3.35               79   
2   J003        M04       Additive           2.29               56   
3   J004        M04       Grinding           1.76              106   
4   J005        M01          Lathe           1.90               46   

   Energy_Consumption  Machine_Availability      Scheduled_Start  \
0               11.42                    96  2023-03-18 08:00:00   
1                6.61                    84  2023-03-18 08:10:00   
2               11.11                    92  2023-03-18 08:20:00   
3               12.50                    95  2023-03-18 08:30:00   
4                8.13                    88  2023-03-18 08:40:00   

         Scheduled_End         Actual_Start           Actual_End Job_Status  \
0  2023-03-18 09:16:00  2023-03-18 08:05:00  2023-03-18 09:21:00  Completed   
1  2023-03-18 09:29:00  2023-03-18 08:20:00  2023-03-18 09:39:00    Delayed   
2  2023-03-18 09:16:00                  NaN                  NaN     Failed   
3  2023-03-18 10:16:00  2023-03-18 08:35:00  2023-03-18 10:21:00  Completed   
4  2023-03-18 09:26:00  2023-03-18 08:42:00  2023-03-18 09:28:00  Completed   

  Optimization_Category  Delay_Minutes  Scheduled_Duration  Actual_Duration  \
0   Moderate Efficiency            5.0                76.0             76.0   
1        Low Efficiency           10.0                79.0             79.0   
2        Low Efficiency            NaN                56.0              NaN   
3   Moderate Efficiency            5.0               106.0            106.0   
4       High Efficiency            2.0                46.0             46.0   

  Scheduled_Date Actual_Date  
0     2023-03-18  2023-03-18  
1     2023-03-18  2023-03-18  
2     2023-03-18         NaN  
3     2023-03-18  2023-03-18  
4     2023-03-18  2023-03-18

# 1. Hypothesis 1:
> **Does Optimization Category affect Energy Consumption?**

## 📌 Hypotheses:

- **Null (H₀):** Mean energy consumption is the same across all optimization categories.

- **Alt (H₁):** At least one optimization category has different average energy consumption.


### ✅ Test: One-way ANOVA

In [7]:
from scipy.stats import f_oneway

# Drop NA just in case
df_clean = df.dropna(subset=['Energy_Consumption', 'Optimization_Category'])

# Get energy consumption by group
groups = [group['Energy_Consumption'].values for name, group in df_clean.groupby('Optimization_Category')]

# Perform ANOVA
f_stat, p_value = f_oneway(*groups)
print("F-statistic:", f_stat)
print("p-value:", p_value)

if p_value < 0.05:
    print("❗ Reject H0 → Optimization Category affects Energy Consumption.")
else:
    print("✅ Fail to Reject H0 → No significant difference in Energy Consumption across categories.")


F-statistic: 100.64938135552423
p-value: 6.563821605534023e-57
❗ Reject H0 → Optimization Category affects Energy Consumption.


### Hypothesis Test Result: Optimization Category vs. Energy Consumption

We performed a **One-Way ANOVA** to test whether the **average energy consumption** differs across different **Optimization Categories**.

> **F-statistic:** 100.65
> **p-value:** 6.56 × 10⁻⁵⁷ (essentially zero)

#### ✅ Interpretation:

Since the p-value is **far below 0.05**, we **reject the null hypothesis.** This means there is a **statistically significant difference** in energy consumption across at least one of the optimization categories.

#### 📌 Conclusion:

> **Optimization strategy has a measurable impact on energy consumption.** This insight suggests that choosing the right optimization category can meaningfully improve energy efficiency in the production process.

# 2. Hypothesis 2:

> Is the average processing time different between Lathe and Grinding operations?

## 📌 Hypotheses:

- **H₀:** μ_lathe = μ_grinding
- **H₁:** μ_lathe ≠ μ_grinding

### ✅ Test: Independent t-test

In [9]:
from scipy.stats import ttest_ind

lathe = df[df['Operation_Type'] == 'Lathe']['Processing_Time'].dropna()
grinding = df[df['Operation_Type'] == 'Grinding']['Processing_Time'].dropna()

t_stat, p_val = ttest_ind(lathe, grinding, equal_var=False)  # Welch's t-test

print("t-statistic:", t_stat)
print("p-value:", p_val)

if p_val < 0.05:
    print("❗ Reject H0 → Processing times differ significantly between Lathe and Grinding.")
else:
    print("✅ Fail to Reject H0 → No significant difference in Processing times.")


t-statistic: -0.7745227650503164
p-value: 0.43906015340704363
✅ Fail to Reject H0 → No significant difference in Processing times.


## 🔍 Hypothesis Test Result: Lathe vs. Grinding – Processing Time

We conducted an independent two-sample t-test to evaluate whether there is a significant difference in average processing time between Lathe and Grinding operations.

- **t-statistic:** -0.77
- **p-value:** 0.439

## ✅ Interpretation:

Since the p-value is **greater than 0.05, we fail to reject the null hypothesis.** This indicates that the observed difference in processing times between Lathe and Grinding operations is not **statistically significant.**

### 📌 Conclusion:

> **Lathe and Grinding operations have similar processing durations**, suggesting that operation type (between these two) does not significantly impact how long a job takes on average.

# 🎯 Hypothesis 3:

> **Is delay more common in a specific machine?**

You can test:

- **Machine_ID vs Delay (binary: delayed or not)**

## ✅ Test: Chi-Square Test of Independence


In [10]:
# Create binary delay flag
df['Delayed'] = df['Delay_Minutes'] > 0

# Contingency table
contingency = pd.crosstab(df['Machine_ID'], df['Delayed'])

from scipy.stats import chi2_contingency
chi2, p, dof, expected = chi2_contingency(contingency)

print("Chi2 Stat:", chi2)
print("p-value:", p)

if p < 0.05:
    print("❗ Reject H0 → Delay is dependent on Machine_ID.")
else:
    print("✅ Fail to Reject H0 → No significant difference in delays by machine.")

Chi2 Stat: 0.9162796333151306
p-value: 0.9222137111311006
✅ Fail to Reject H0 → No significant difference in delays by machine.


## Hypothesis Test Result: Machine_ID vs. Job Delay

We performed a **Chi-Square Test of Independence** to determine whether **job delays** are associated with specific **machines.**

- Chi² Statistic: 0.92

- p-value: 0.922

## ✅ Interpretation:

Since the p-value is **much greater than 0.05, we fail to reject the null hypothesis.** This means there is **no statistically significant relationship** between the machine used and whether a job is delayed.

## 📌 Conclusion:

> **Delays appear to be evenly distributed across machines,** suggesting that machine choice alone does **not significantly influence** the likelihood of delay in job completion.

# Correlation Test

> Is Energy Consumption correlated with Processing Time?

In [12]:
from scipy.stats import pearsonr

x = df['Processing_Time']
y = df['Energy_Consumption']

corr, p_val = pearsonr(x, y)
print("Pearson Correlation:", corr)
print("p-value:", p_val)

if p_val < 0.05:
    print("❗ Significant correlation exists.")
else:
    print("✅ No significant correlation.")

Pearson Correlation: 0.00037139609020142225
p-value: 0.9906411190171263
✅ No significant correlation.


## 🔍 Correlation Analysis: Processing Time vs. Energy Consumption

We used **Pearson’s correlation test** to examine the linear relationship between **Processing Time and Energy Consumption**.

- **Correlation coefficient:** 0.00037
- **p-value:** 0.9906

## ✅ Interpretation:

The correlation coefficient is **very close to zero,** and the p-value is **much greater than 0.05**, indicating **no statistically significant linear relationship** between processing time and energy usage.

## 📌 Conclusion:

> **Processing time does not influence energy consumption in a linear way**. This suggests that energy usage may be affected more by other factors, such as operation type, material, or optimization strategy.